aixcalibuha/sensitivity_analysis/sobol.py

```diff
@@ -68,13 +85,301 @@ def generate_samples(self):
+        figs, axes = SenAnalyzer.plot_single(
+            result=result,
+            show_plot=False,
+            cal_classes=cal_classes,
+            goals=goals,
+            figs_axes=figs_axes,
+            use_suffix=use_suffix
+        )
+        # set new title for the figures of each calibration class
+        for fig in figs:
+            fig.suptitle(f"Interaction of {para_name} in class {fig._suptitle.get_text()}")
+        if show_plot:
+            plt.show()
+        return figs, axes
+
+    @staticmethod
+    def heatmap(result, cal_class, goal, ax=None, show_plot=True, use_suffix=False):
+        """
+        Plot S2 sensitivity results from one calibration class and goal as a heatmap.
+
+        :param pd.DataFrame result:
+            A second order result from run
+        :param str cal_class:
+            Name of the class to plot S2 from.
+        :param str goal:
+            Name of the goal to plot S2 from.
+        :param matplotlib.axes ax:
+            Default is None. If an axes is given the heatmap will be plotted on it else
+            a new figure and axes is created.
+        :param bool show_plot:
+            Default is True. If False, all created plots are not shown.
+        :param bool use_suffix:
+            Default is False. If True, only the last suffix of a Modelica variable is displayed.
+        :return:
+            Returns axes
+        """
+        if use_suffix:
+            result = _rename_tuner_names(result)
+        if ax is None:
+            fig, ax = plt.subplots()
+        data = result.sort_index().loc[cal_class, goal, 'S2'].fillna(0).reindex(
+            index=result.columns)
+        im = ax.imshow(data, cmap='Reds')
+        ax.set_title(f'Class: {cal_class} Goal: {goal}')
+        ax.set_xticks(np.arange(len(data.columns)))
+        ax.set_yticks(np.arange(len(data.index)))
+        ax.set_xticklabels(data.columns)
+        ax.set_yticklabels(data.index)
+        ax.spines[:].set_color('black')
+        plt.setp(ax.get_xticklabels(), rotation=90, ha="right", rotation_mode="anchor")
+        cbar = ax.figure.colorbar(im, ax=ax)
+        cbar.ax.set_ylabel("S2", rotation=90)
+        if show_plot:
+            plt.show()
+        return ax
+
+    @staticmethod
+    def heatmaps(result, **kwargs):
+        """
+        Plot S2 sensitivity results as a heatmap for multiple
+        calibration classes and goals in one figure.
+
+        :param pd.DataFrame result:
+            A second order result from run
+        :keyword [str] cal_class:
+            Default is a list of all calibration classes in the result.
+            If a list of classes is given only these classes are plotted.
+        :keyword [str] goal:
+            Default is a list of all goals in the result.
+            If a list of goals is given only these goals are plotted.
+        :keyword bool show_plot:
+            Default is True. If False, all created plots are not shown.
```

Comment:
A general note for all plot functions: Having the keyword arguments as follows helps coding IDE like PyCharm to hint at correct parameters. Furthermore, passing a kwargs dict still works. I try to use **kwargs only if there are many arguments or if the structure of the function is not clear for inheritance.
```
def heatmaps(result: pd.DataFrame, cal_class: str = None, goals: str = None, show_plot: bool = True):
```
Also, goal should be goals in this case.